daemon/cmd/daemon_main.go

```diff
@@ -1482,6 +1485,12 @@ func runDaemon() {
 		d.SendNotification(monitorAPI.AgentNotifyStart, repr)
 	}
 
+	// clean up all arp PERM entries that might have previously set by
+	// a Cilium instance
+	if !d.datapath.Node().NodeNeighDiscoveryEnabled() || option.Config.GCNeighEntries {
```

Comment:
Do we still want to keep the cilium-agent param? My suggestion was to remove the stored neigh link after the removal which would indicate that the removal was successful.

Reply:
I thought about it and I don't think it hurts to leave it as a potential fallback.

Reply:
My previous comment got lost :S Anyway, IMHO yet another flag hinders Cilium UX. If we can do it without a flag, then we should do it. Especially considering that now we have a reliable mechanism which eliminates the need for the flag. Another thing, if a user sets a flag while the neigh discovery is on, then we might remove valid entries + remove the link device which will make it impossible to remove the entries in the future without relying on the flag. Finally, removing the flag makes the change smaller.